### ISSUE TITLE
Ubuntu Server Ps3NetSrv?

### ISSUE THREAD

**Issue by IllMethods** [aldostools/webMAN-MOD#315]
Hello

I've seen some people using Ps3NetSrv on Ubuntu but not sure if on a headless server? Is this a thing or not? Also, if it isn't could it be? Use a web based config?

Also, just wondering if loading ps2 isos like ps1 isos not needing a transfer and also the icons working properly is something you are planning on working on. I'm happy this works but this would just make it slightly better.

**Comment by shawly** [comment]
You can run it headless, I recommend using docker because it's easy to set up.
I also created a container with the latest binary built from source: https://hub.docker.com/r/shawly/ps3netsrv

**Edit:** For anyone interested, I've been able to add multiarch support, people can now run the container on all RaspberryPi models or stuff like Synology disk stations. This hasn't been tested properly as I only own an RPi 1B+, which works but is really slow (obviously).

**Comment by IllMethods** [comment]
I figured it out but I couldn't build the latest version, didn't have everything required so just built an older version. Also never been a user of docker, usually just create scripts to run all my apps without a dock.

**Comment by shawly** [comment]
My container runs the latest version built from master, with docker you don't have to compile stuff yourself.
You're surely missing out, it's worth a try and you'll probably never go back if you've used Docker once for running applications. :-)

**Comment by IllMethods** [comment]
Does docker work like a vm?

**Comment by shawly** [comment]
From a technical standpoint? No, it's an isolated environment for running applications. Containers run on the same kernel as your host's OS.
It's like running the application on your host but it is self contained so you don't need to install any dependencies (other than docker) and you don't have to bother with compilation issues and stuff. You can run a CentOS container on an Ubuntu host and vice versa. 
This keeps your computer clean from thousands of build libraries. And if you delete your container and the image there is no trace left of it ever being "installed" on your system.
Compared to a VM there is almost no overhead because it's not virtualized so it's easy on resources.

Backblaze has a pretty nice article which explains the differences better than I do: https://www.backblaze.com/blog/vm-vs-containers/

You can see a container more like an already "preconfigured vm" (just without virtualization and the heavier resource usage). My ps3netsrv container has roughly the size of 10MB and is installed by running a single command, a vm would at least take up 3-4GB of space and would require you to set up everything manually.

**Comment by IllMethods** [comment]
Hmmm, ok. I might give it a jam next time I have to do a full reinstall of os. I already have too many things installed and it's running perfectly so not keen on moving them all around, but I know a lot of them have docks already made for.

**Comment by shawly** [comment]
For anyone interested, I've been able to add multiarch support, people can now run the container on all RaspberryPi models or stuff like Synology disk stations. This hasn't been tested properly as I only own an RPi 1B+, which works but is really slow (obviously).

https://hub.docker.com/r/shawly/ps3netsrv

**Comment by IllMethods** [comment]
Just curious, can you have it use more than one dir?

Say run 

./ps3netsrv '/blahblahblah' '/blahblahblah2'

???

**Comment by aldostools** [comment]
No, but you could create symbolic links to external files or secondary folders in the shared folder.

$ ln -s /path/to/dir /path/to/symlink

**Comment by IllMethods** [comment]
Ok cool, and the symlink would still show both dirs as one?

I'm more a windows person but have been running ubuntu server for years for a media server, I know the basics but some things I'm not too sure on. 

Also, do docks auto update?

**Comment by aldostools** [comment]
Symlinks to other directories would show as subdirectories of the shared folder.
Symlinks or hard links to files would show as regular files. e.g. you can link ISOs, videos, pkg files, etc.

The **ln** command also exists on Windows 10 as [MKLINK.EXE](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/mklink). Run it using CMD 

[Watchtower](https://hub.docker.com/r/containrrr/watchtower) – Automatically Update Running Docker Containers.
https://www.ostechnix.com/automatically-update-running-docker-containers/

**Comment by Joonie86** [comment]
No further feedback
